### PR TITLE
Web Editor: Scripts out of process execution.

### PR DESCRIPTION
**Pull request by Kuinox**

### CODE REVIEW

**Comment by Kuinox** [comment]
Fixes #71

File: src/Draco.Editor.Web/Program.cs [hunk 1]
```diff
@@ -89,28 +86,41 @@ private static async Task ProcessUserInput()
...
+            if (!processedAssemblyNames.Contains(currentAssembly))
+            {
+                processedAssemblyNames.Add(currentAssembly);
+                foreach (var referencedAssembly in currentAssembly.GetReferencedAssemblies())
+                {
+                    assembliesToProcess.Enqueue(Assembly.Load(referencedAssembly));
+                }
+            }
+        }
+        return processedAssemblyNames.Where(s => s != assembly).Select(s => s.GetName().Name+".dll")!;
```
Comment (LPeter1997) [inline src/Draco.Editor.Web/Program.cs:106]: Space around `+`, or even better, interpolation

File: src/Draco.Editor.Web/Program.cs [hunk 2]
```diff
@@ -89,28 +86,41 @@ private static async Task ProcessUserInput()
...
         try
         {
-            var execResult = ScriptingEngine.Execute(compilation);
-            if (!execResult.Success) outputText = string.Join("\n", execResult.Diagnostics);
+            var stream = GetAssemblyStream(compilation);
+            if (stream is null) return;
+            var buffer = stream.ToArray();
+            var assembly = Assembly.Load(buffer);
+            var references = GetAllReferencedAssemblyNames(assembly);
+            SendAssembly(assembly.GetName().Name! + ".dll", buffer, references.ToArray());
```
Comment (LPeter1997) [inline src/Draco.Editor.Web/Program.cs:118]: At least plus is padded, do we want interpolation here?

File: src/Draco.Editor.Web/Program.cs [hunk 3]
```diff
@@ -166,20 +180,33 @@ private static void SetOutputText(string text)
...
+                    buffer = (byte[]?)default
+                }).Append(
+                    new
+                    {
+                        behavior = "assembly",
+                        name = assemblyName,
+                        buffer = (byte[]?)assembly
+                    }
+                ).Append(
+                    new
```
Comment (LPeter1997) [inline src/Draco.Editor.Web/Program.cs:203]: Why not inline the new? One less indentation

File: src/Draco.Editor.Web/Program.cs [hunk 4]
```diff
@@ -166,20 +180,33 @@ private static void SetOutputText(string text)
...
-        }
+                assemblyRootFolder = "_framework", //TODO: this value should be controlled by the JS
+                mainAssemblyName = assemblyName,
+                assets = referencedAssemblies.Select(s => new
+                {
+                    behavior = "assembly",
+                    name = s,
+                    buffer = (byte[]?)default
+                }).Append(
+                    new
```
Comment (LPeter1997) [inline src/Draco.Editor.Web/Program.cs:196]: Why not inline the new? One less indentation

File: src/Draco.Editor.Web/app/ts/app.ts [hunk 5]
```diff
@@ -5,12 +5,14 @@ import { Registry } from 'monaco-textmate';
...
 import { deflateRaw, inflateRaw } from 'pako';
+import { stdout } from 'process';
 
 // This file is run on page load.
 // This run before blazor load, and will tell blazor to start.
 
-const worker = new Worker('worker.js'); // first thing: we start the worker so it loads in parallel.
-
+const compilerWorker = new Worker('worker.js'); // first thing: we start the worker so it loads in parallel.
+let runtimeWorker : Worker | undefined;
```
Comment (LPeter1997) [inline src/Draco.Editor.Web/app/ts/app.ts:14]: Isn't the convention `name: type` and not `name : type`?

File: src/Draco.Editor.Web/app/ts/worker.ts [hunk 6]
```diff
@@ -61,27 +58,40 @@ function sendMessage(type: string, message: string) {
...
-        }
-    );
-    const exports = await getAssemblyExports(monoCfg['mainAssemblyName']);
-    csOnMessage = exports.Draco.Editor.Web.Interop.OnMessage;
+        );
+        const exports = await getAssemblyExports(monoCfg['mainAssemblyName']);
+        csOnMessage = exports.Draco.Editor.Web.Interop.OnMessage;
+    }
+
+
```
Comment (LPeter1997) [inline src/Draco.Editor.Web/app/ts/worker.ts:94]: DOuble blank line

File: src/Draco.Editor.Web/app/ts/app.ts (hunk 5 above)
Comment (Kuinox) [reply src/Draco.Editor.Web/app/ts/app.ts:14]: Yeah, autoformat is breaking the linter convention, was too lazy to spend time fixing it so no autoformating ^^'.

**Review by LPeter1997 — APPROVED** [review]
LGTM

**Pull request merged**